ycmd/completers/typescript/typescript_completer.py

```diff
@@ -46,16 +49,17 @@ def __init__( self, user_options ):
 
     binarypath = utils.PathToFirstExistingExecutable( [ 'tsserver' ] )
     if not binarypath:
-      raise RuntimeError(
-        'tsserver not found. TypeScript 1.5 or higher is required' )
+      _logger.error( BINARY_NOT_FOUND_MESSAGE )
+      raise RuntimeError( BINARY_NOT_FOUND_MESSAGE )
 
     # Each request sent to tsserver must have a sequence id.
     # Responses contain the id sent in the corresponding request.
     self._sequenceid = 0
     self._tsserver_handle = utils.SafePopen( binarypath,
-      stdout = subprocess.PIPE,
-      stdin  = subprocess.PIPE
-    )
+                                             stdout = subprocess.PIPE,
+                                             stdin = subprocess.PIPE,
+                                             stderr = subprocess.STDOUT,
+                                             universal_newlines = True )
```

Comment:
After investigation, the issue is that the value of `Content-Length` in the header is one off. What happens is:
  - first response is read until the character `\r`. 
  - second response reads the left character `\n` and fails.

The option `universal_newlines` solves this because the two characters `\r\n` are converted to one `\n`. Another solution would be to always add one to the contents length.